security/user-last-seen.php

```diff
@@ -0,0 +1,273 @@
+		$admin_notices_hook_name = is_network_admin() ? 'network_admin_notices' : 'admin_notices';
+
+		if ( isset( $_GET['reset_last_seen_success'] ) && '1' === $_GET['reset_last_seen_success'] ) {
+			add_action( $admin_notices_hook_name, function() {
+				$class = 'notice notice-success is-dismissible';
+				$error = __( 'User unblocked.', 'inactive-account-lockdown' );
+
+				printf( '<div class="%1$s"><p>%2$s</p></div>', esc_attr( $class ), esc_html( $error ) );
+			} );
+		}
+
+		if ( ! isset( $_GET['user_id'] ) || ! isset( $_GET['action'] ) || 'reset_last_seen' !== $_GET['action'] ) {
+			return;
+		}
+
+		$user_id = absint( $_GET['user_id'] );
+
+		$error = null;
+		if ( ! isset( $_GET['reset_last_seen_nonce'] ) || ! wp_verify_nonce( sanitize_text_field( $_GET['reset_last_seen_nonce'] ), 'reset_last_seen_action' ) ) {
+			$error = __( 'Unable to verify your request', 'inactive-account-lockdown' );
+		}
+
+		if ( ! get_userdata( $user_id ) ) {
+			$error = __( 'User not found.', 'inactive-account-lockdown' );
+		}
+
+		if ( ! current_user_can( 'edit_user', $user_id ) ) {
+			$error = __( 'You do not have permission to unblock this user.', 'inactive-account-lockdown' );
+		}
+
+		if ( ! $error && ! delete_user_meta( $user_id, self::LAST_SEEN_META_KEY ) ) {
+			$error = __( 'Unable to unblock user.', 'inactive-account-lockdown' );
+		}
+
+		if ( $error ) {
+			add_action( $admin_notices_hook_name, function() use ( $error ) {
+				$class = 'notice notice-error is-dismissible';
+
+				printf( '<div class="%1$s"><p>%2$s</p></div>', esc_attr( $class ), esc_html( $error ) );
+			} );
+			return;
+		}
+
+		$url = remove_query_arg( array(
+			'action',
+			'user_id',
+			'reset_last_seen_nonce',
+		) );
+
+		$url = add_query_arg( array(
+			'reset_last_seen_success' => 1,
+		), $url );
+
+		wp_safe_redirect( $url );
+		exit();
+	}
+
+	public function register_release_date() {
+		if ( ! get_option( self::LAST_SEEN_RELEASE_DATE_TIMESTAMP_OPTION_KEY ) ) {
+			// Right after the first admin_init, set the release date timestamp
+			// to be used as a fallback for users that never logged in before.
+			add_option( self::LAST_SEEN_RELEASE_DATE_TIMESTAMP_OPTION_KEY, time() );
+		}
+	}
+
+	public function is_considered_inactive( $user_id ) {
+		$last_seen_timestamp = get_user_meta( $user_id, self::LAST_SEEN_META_KEY, true );
+		if ( $last_seen_timestamp ) {
+			return $last_seen_timestamp < $this->get_inactivity_timestamp();
+		}
+
+		$release_date_timestamp = get_option( self::LAST_SEEN_RELEASE_DATE_TIMESTAMP_OPTION_KEY );
+		if ( $release_date_timestamp ) {
+			return $release_date_timestamp < $this->get_inactivity_timestamp();
+		}
+
+		// Release date is not defined yed, so we can't consider the user inactive.
+		return false;
+	}
+
+	public function get_inactivity_timestamp() {
+		return strtotime( sprintf( '-%d days', constant( 'VIP_SECURITY_CONSIDER_USERS_INACTIVE_AFTER_DAYS' ) ) ) + self::LAST_SEEN_UPDATE_USER_META_CACHE_TTL;
```

Comment:
Do we need to validate/sanitize `VIP_SECURITY_CONSIDER_USERS_INACTIVE_AFTER_DAYS` first here? Make sure it's a number string and if not fallback to some sane default?